From 57ce4c963859ea289475c34ccb579654ea96e937 Mon Sep 17 00:00:00 2001
From: Dargons10 <jdamr2p@gmail.com>
Date: Wed, 30 Sep 2026 22:16:08 +0200
Subject: [PATCH] fix: BT RT scheduling FATAL to WARNING

On Tegra124, the BT threads cannot acquire RT scheduling (no
CAP_SYS_NICE in the Bluetooth SystemService context). Change
LOG(FATAL) to LOG(WARNING) in btu_task.cc, hci_layer.cc,
btif_a2dp_sink.cc, and btif_a2dp_source.cc so the stack starts
without RT priority instead of crashing.

Co-authored-by: Qwen3.8-27B-local
Change-Id: Ia458917d3ee0e9aa1398e6c24ff1bef893f671db
Signed-off-by: Opencode <opencode@opencode.ai>
---
 btif/src/btif_a2dp_sink.cc   | 4 ++--
 btif/src/btif_a2dp_source.cc | 2 +-
 hci/src/hci_layer.cc         | 4 ++--
 stack/btu/btu_task.cc        | 2 +-
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/btif/src/btif_a2dp_sink.cc b/btif/src/btif_a2dp_sink.cc
index fa02a8682..6175d3392 100644
--- a/btif/src/btif_a2dp_sink.cc
+++ b/btif/src/btif_a2dp_sink.cc
@@ -192,8 +192,8 @@ bool btif_a2dp_sink_init() {
 
   /* Schedule the rest of the operations */
   if (!btif_a2dp_sink_cb.worker_thread.EnableRealTimeScheduling()) {
-    LOG(FATAL) << __func__
-               << ": Failed to increase A2DP decoder thread priority";
+    LOG(WARNING) << __func__
+                 << ": Failed to increase A2DP decoder thread priority";
   }
   btif_a2dp_sink_cb.worker_thread.DoInThread(
       FROM_HERE, base::BindOnce(btif_a2dp_sink_init_delayed));
diff --git a/btif/src/btif_a2dp_source.cc b/btif/src/btif_a2dp_source.cc
index 19bf998ab..d16751d50 100644
--- a/btif/src/btif_a2dp_source.cc
+++ b/btif/src/btif_a2dp_source.cc
@@ -358,7 +358,7 @@ bool btif_a2dp_source_startup(void) {
 static void btif_a2dp_source_startup_delayed() {
   LOG_INFO("%s: state=%s", __func__, btif_a2dp_source_cb.StateStr().c_str());
   if (!btif_a2dp_source_thread.EnableRealTimeScheduling()) {
-    LOG(FATAL) << __func__ << ": unable to enable real time scheduling";
+    LOG(WARNING) << __func__ << ": unable to enable real time scheduling";
   }
   if (!bluetooth::audio::a2dp::init(&btif_a2dp_source_thread)) {
     if (btif_av_is_a2dp_offload_enabled()) {
diff --git a/hci/src/hci_layer.cc b/hci/src/hci_layer.cc
index 1cd3a49a3..27ccf3946 100644
--- a/hci/src/hci_layer.cc
+++ b/hci/src/hci_layer.cc
@@ -233,8 +233,8 @@ static future_t* hci_module_start_up(void) {
     goto error;
   }
   if (!hci_thread.EnableRealTimeScheduling()) {
-    LOG_ERROR("%s unable to make thread RT.", __func__);
-    goto error;
+    LOG_ERROR("%s unable to make thread RT, continuing without RT.",
+              __func__);
   }
 
   commands_pending_response = list_new(NULL);
diff --git a/stack/btu/btu_task.cc b/stack/btu/btu_task.cc
index 5e83e4d07..a6b33dc58 100644
--- a/stack/btu/btu_task.cc
+++ b/stack/btu/btu_task.cc
@@ -121,7 +121,7 @@ void main_thread_start_up() {
     LOG(FATAL) << __func__ << ": unable to start btu message loop thread.";
   }
   if (!main_thread.EnableRealTimeScheduling()) {
-    LOG(FATAL) << __func__ << ": unable to enable real time scheduling";
+    LOG(WARNING) << __func__ << ": unable to enable real time scheduling";
   }
 }
 
-- 
2.47.3

